Sources/Web5/Dids/Methods/DIDDHT.swift

```diff
@@ -0,0 +1,433 @@
+                    for (k, v) in answer.attributes {
+                        let splitValue = v.components(separatedBy: DIDDHT.Constants.PROPERTY_SEPARATOR)
+                        answer.attributes[k] = splitValue[0]
+
+                        for i in 1..<splitValue.count {
+                            let parts = splitValue[i].components(separatedBy: "=")
+                            if parts.count == 2 {
+                                answer.attributes[parts[0]] = parts[1]
+                            }
+                        }
+                    }
+
+                    return answer
+                } else {
+                    return answer
+                }
+            }
+
+            return message
+        }
+
+        /// Converts a DNS packet into a DID Document and its associated metadata,
+        /// according to the `did:dht` [spec](https://tbd54566975.github.io/did-dht-method/#dids-as-a-dns-packet)
+        static func fromDNSPacket(
+            did: DID,
+            dnsPacket: DNS.Message
+        ) throws -> (DIDDocument, DIDDocument.Metadata) {
+            var idLookup = [String: String]()
+
+            var alsoKnownAs = [String]()
+            var controllers = [String]()
+            var verificationMethods = [VerificationMethod]()
+            var services = [Service]()
+            var authentication: [EmbeddedOrReferencedVerificationMethod]?
+            var assertionMethod: [EmbeddedOrReferencedVerificationMethod]?
+            var capabilityDelegation: [EmbeddedOrReferencedVerificationMethod]?
+            var capabilityInvocation: [EmbeddedOrReferencedVerificationMethod]?
+            var keyAgreement: [EmbeddedOrReferencedVerificationMethod]?
+            var types: [Int]?
+
+            /// `did:dht` properties are ONLY present in DNS TXT records.
+            /// Loop through the answers, only taking into consideration the text records.
+            for case let answer as TextRecord in dnsPacket.answers {
+                let dnsRecordID = String(answer.name.prefix { $0 != "." }.dropFirst())
+
+                if dnsRecordID.hasPrefix("aka") {
+                    // Process an also known as record
+                    alsoKnownAs.append(contentsOf: answer.values)
+                } else if dnsRecordID.hasPrefix("cnt") {
+                    // Process a controller record
+                    controllers.append(contentsOf: answer.values)
+                } else if dnsRecordID.hasPrefix("k") {
+                    // Process verification methods
+                    let publicKeyBytes = try answer.attributes["k"]!.decodeBase64Url()
+                    let namedCurve = RegisteredKeyType(rawValue: Int(answer.attributes["t"]!)!)
+
+                    let publicKey: Jwk
+                    switch namedCurve {
+                    case .Ed25519:
+                        publicKey = try EdDSA.Ed25519.publicKeyFromBytes(publicKeyBytes)
+                    case .secp256k1:
```

Comment:
will secp256r1 be added later?

Reply:
Yup! Currently just isn't implemented yet, so haven't added it in quite yet.